### DISCUSSION TITLE
Is Url.AbsoluteContent extension method still available in version 5?

### DISCUSSION THREAD

**Discussion by xperiandri** [ASP-NET-MVC-Boilerplate/Framework#122] [Q&A]
I can't find those extension methods

**Comment by RehanSaeed** [comment]
I recommend switching to use `LinkGenerator` now. See my answer in https://stackoverflow.com/questions/30755827/getting-absolute-urls-using-asp-net-core.

**Comment by xperiandri** [comment]
Then this change is what you expect, isn't it?
``` C#
// OLD
new Uri(Url.AbsoluteContent($"~/img/og-image.jpg?v={favIconVersion}")),
// NEW
new UriBuilder(request.Scheme, 
               request.Host.Host, 
               request.Host.Port.GetValueOrDefault(), 
               Url.Content($"~/img/og-image.jpg?v={favIconVersion}"))
    .Uri,
```

**Comment by RehanSaeed** [comment]
Hmmmm, this is a good question and I can't see an answer for how to do this using `LinkGenerator`. I suggest asking this question at the bottom of this docs page:

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-2.1

Reopening until we get an answer.

**Comment by RehanSaeed** [comment]
This is now answered in https://github.com/dotnet/AspNetCore.Docs/issues/16501.
